cmapPy/pandasGEXpress/concat_gctoo.py

```diff
@@ -261,16 +262,16 @@ def assemble_common_meta(common_meta_dfs, fields_to_remove):
         all_meta_df_sorted (pandas df)
 
     """
-    # Remove any column headers that will prevent dfs from being identical
-    for df in common_meta_dfs:
-        df.drop(fields_to_remove, axis=1, inplace=True)
-
     # Remove any column headers that are not present in all dfs (and sort)
     shared_column_headers = sorted(set.intersection(*[set(df.columns) for df in common_meta_dfs]))
-    common_meta_dfs = [df[shared_column_headers] for df in common_meta_dfs]
+    trimmed_common_meta_dfs = [df[shared_column_headers] for df in common_meta_dfs]
+
+    # Remove any column headers that will prevent dfs from being identical
```

Comment:
Previously, this code modified the input matrices in-place, which wasn't ideal and annoying for testing. SO that's what trimmed_common_meta_dfs is about. I think you're right that even_more_trimmed_common_meta_dfs is probably overkill -- we can modify trimmed_common_meta_dfs in-place without trouble.